### PR TITLE
add log file line number to Issue data.

### PR DESCRIPTION
**Pull request by TingluoHuang**

### CODE REVIEW

**Comment by TingluoHuang** [comment]
image


image

File: src/Agent.Worker/TaskCommandExtension.cs [hunk 1]
```diff
@@ -359,7 +347,7 @@ private Issue CreateIssue(IExecutionContext context, string issueType, String me
                 throw new Exception($"issue type {issueType} is not an expected issue type.");
             }
 
-            messageToLog = message;
+            string messageToLog = message;
```
Comment (ericsciple) [inline src/Agent.Worker/TaskCommandExtension.cs]: ?

File: src/Microsoft.VisualStudio.Services.Agent/Logging.cs [hunk 2]
```diff
@@ -63,6 +68,7 @@ public void Write(string message)
 
             string line = $"{DateTime.UtcNow.ToString("O")} {message}";
             _pageWriter.WriteLine(line);
+            _totalLines++;
```
Comment (ericsciple) [inline src/Microsoft.VisualStudio.Services.Agent/Logging.cs]: count new lines?

File: src/Agent.Worker/ExecutionContext.cs [hunk 3]
```diff
@@ -441,12 +458,14 @@ public void InitializeJob(JobRequestMessage message, CancellationToken token)
...
         // the rule is command messages - which should be crafted using strongly typed wrapper methods.
-        public void Write(string tag, string message)
+        public long Write(string tag, string message)
         {
             string msg = _secretMasker.MaskSecrets($"{tag}{message}");
+            long totalLines;
             lock (_loggerLock)
             {
                 _logger.Write(msg);
+                totalLines = _logger.TotalLines;
```
Comment (ericsciple) [inline src/Agent.Worker/ExecutionContext.cs]: switch order? record _logger.TotalLines + 1 then write?

**Pull request merged**